crates/re_space_view_dataframe/src/space_view_class.rs

```diff
@@ -0,0 +1,190 @@
+use crate::view_part_system::EmptySystem;
+use egui_extras::Column;
+use itertools::Itertools;
+use re_data_store::{EntityProperties, InstancePath};
+use re_log_types::EntityPath;
+use re_query::get_component_with_instances;
+use re_viewer_context::{
+    AutoSpawnHeuristic, PerSystemEntities, SpaceViewClass, SpaceViewClassRegistryError,
+    SpaceViewId, SpaceViewSystemExecutionError, SystemExecutionOutput, UiVerbosity, ViewQuery,
+    ViewerContext,
+};
+
+#[derive(Default)]
+pub struct DataframeSpaceView;
+
+impl SpaceViewClass for DataframeSpaceView {
+    type State = ();
+
+    const IDENTIFIER: &'static str = "Dataframe";
+    const DISPLAY_NAME: &'static str = "Dataframe";
+
+    fn icon(&self) -> &'static re_ui::Icon {
+        //TODO(ab): fix that icon
+        &re_ui::icons::SPACE_VIEW_TEXTBOX
+    }
+
+    fn help_text(&self, _re_ui: &re_ui::ReUi) -> egui::WidgetText {
+        "Show the data contained in entities in a table.".into()
+    }
+
+    fn on_register(
+        &self,
+        system_registry: &mut re_viewer_context::SpaceViewSystemRegistry,
+    ) -> Result<(), SpaceViewClassRegistryError> {
+        system_registry.register_part_system::<EmptySystem>()
+    }
+
+    fn preferred_tile_aspect_ratio(&self, _state: &Self::State) -> Option<f32> {
+        None
+    }
+
+    fn layout_priority(&self) -> re_viewer_context::SpaceViewClassLayoutPriority {
+        re_viewer_context::SpaceViewClassLayoutPriority::Low
+    }
+
+    fn auto_spawn_heuristic(
+        &self,
+        _ctx: &ViewerContext<'_>,
+        _space_origin: &EntityPath,
+        _ent_paths: &PerSystemEntities,
+    ) -> re_viewer_context::AutoSpawnHeuristic {
+        AutoSpawnHeuristic::NeverSpawn
+    }
+
+    fn selection_ui(
+        &self,
+        _ctx: &ViewerContext<'_>,
+        _ui: &mut egui::Ui,
+        _state: &mut Self::State,
+        _space_origin: &EntityPath,
+        _space_view_id: SpaceViewId,
+        _root_entity_properties: &mut EntityProperties,
+    ) {
+    }
+
+    fn ui(
+        &self,
+        ctx: &ViewerContext<'_>,
+        ui: &mut egui::Ui,
+        _state: &mut Self::State,
+        _root_entity_properties: &EntityProperties,
+        query: &ViewQuery<'_>,
+        _system_output: SystemExecutionOutput,
+    ) -> Result<(), SpaceViewSystemExecutionError> {
+        re_tracing::profile_function!();
+
+        let entities: Vec<_> = query
+            .iter_all_data_results()
+            .filter(|data_result| data_result.resolved_properties.visible)
+            .map(|data_result| &data_result.entity_path)
+            .unique()
+            .cloned()
+            .collect();
```

Comment:
Does `iter_all_data_results` guarantee the order of the entities?
I would expect to have `entities` sorted, but I can't tell if that's what we get.

Reply:
todo: btreeset

Reply:
Moved everything I could to btreeset. We have one remaining vec for the rows, since I need random access.